web/impact/impact/tests/test_mentor_program_office_hour_list_view.py

```diff
@@ -0,0 +1,164 @@
+        self._check_response_values(
+            {'mentor_name': mentor.full_name()},
+            mentor_office_hours
+        )
+
+    def test_nonexistent_mentor_id_returns_nothing(self):
+        mentor, mentor_office_hours = self._create_user_office_hours()
+        self._assert_nil_response_values(
+            {'mentor_id': NON_EXISTENT_ID}
+        )
+
+    def test_nonexistent_mentor_name_returns_nothing(self):
+        mentor, mentor_office_hours = self._create_user_office_hours()
+        self._assert_nil_response_values(
+            {'mentor_name': NON_EXISTENT_NAME}
+        )
+
+    def test_filters_by_finalist_id(self):
+        finalist, finalist_office_hours = self._create_user_office_hours(
+            mentor=False
+        )
+        self._check_response_values(
+            {'finalist_id': finalist.id},
+            finalist_office_hours
+        )
+
+    def test_filters_by_finalist_name(self):
+        finalist, finalist_office_hours = self._create_user_office_hours(
+            mentor=False
+        )
+        self._check_response_values(
+            {'finalist_name': finalist.full_name()},
+            finalist_office_hours
+        )
+
+    def test_filters_by_partial_finalist_name(self):
+        finalist, finalist_office_hours = self._create_user_office_hours(
+            mentor=False
+        )
+        self._check_response_values(
+            {'finalist_name': finalist.first_name},
+            finalist_office_hours
+        )
+
+    def test_nonexistent_finalist_id_returns_nothing(self):
+        finalist, finalist_office_hours = self._create_user_office_hours(
+            mentor=False
+        )
+        self._assert_nil_response_values(
+            {'finalist_id': NON_EXISTENT_ID}
+        )
+
+    def test_nonexistent_finalist_name_returns_nothing(self):
+        finalist, finalist_office_hours = self._create_user_office_hours(
+            mentor=False
+        )
+        self._assert_nil_response_values(
+            {'finalist_name': NON_EXISTENT_NAME}
+        )
+
+    def _check_response_values(self, params, office_hours):
+        response = self._get_response_as_logged_in_user(params)
+        self.assertEqual(response.data["count"], USER_COUNT)
+        self.assertTrue(self._compare_ids(office_hours, response))
+
+    def _assert_nil_response_values(self, params):
+        response = self._get_response_as_logged_in_user(params)
+        self.assertEqual(response.data["count"], 0)
+        self.assertEqual(response.data["results"], [])
+
+    def _create_user_office_hours(self, mentor=True):
+        if mentor:
+            user = ExpertFactory()
+            office_hours = MentorProgramOfficeHourFactory.create_batch(
+                USER_COUNT, mentor=user)
+        else:
+            user = EntrepreneurFactory()
+            office_hours = MentorProgramOfficeHourFactory.create_batch(
+                USER_COUNT, finalist=user)
+        return user, office_hours
```

Comment:
you could probably get rid of the conditional by passing the factory as a function argument

Reply:
we wouldn't avoid the conditional as the arguments for `create_batch` correspond to fields in the MentorProgramOfficeHour model. We'd still have to either set the `finalist` or `mentor` fields

Reply:
we could probably achieve that by passing in a dict that we expand within the function e.g.
```
    def _create_user_office_hours(self, Factory, user_dict):
        user = Factory()
        office_hours = MentorProgramOfficeHourFactory.create_batch(
            USER_OFFICE_HOUR_COUNT, **user_dict)
        return user, office_hours
```

then we would use it like so
```
user, office_hours = self. _create_user_office_hours(ExpertFactory, {"mentor": user})
```

Reply:
@sante in this case we'd still have to instantiate the user object from outside and pass it in as an argument

Reply:
oh I see, that code snippet actually doesn't make sense 😆